src/plugins/intel_cpu/src/mkldnn_plugin.cpp

```diff
@@ -663,22 +757,76 @@ Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std
+bool Engine::isLegacyAPI() const {
+    const auto& core = GetCore();
+    if (!core)
+        IE_THROW() << "Unable to get API version. Core is unavailable";
+
+    return !core->isNewAPI();
+}
+
+Parameter Engine::GetConfigLegacy(const std::string& name, const std::map<std::string, Parameter>& options) const {
     Parameter result;
     auto option = engConfig._config.find(name);
     if (option != engConfig._config.end()) {
         result = option->second;
     } else {
-        IE_THROW() << "Unsupported config key " << name;
+        IE_THROW() << "Unsupported config parameter: " << name;
     }
     return result;
 }
 
+Parameter Engine::GetConfig(const std::string& name, const std::map<std::string, Parameter>& options) const {
+    if (isLegacyAPI())
+        return GetConfigLegacy(name, options);
+
+    if (name == ov::optimal_number_of_infer_requests) {
+        const auto streams = engConfig.streamExecutorConfig._streams;
+        return static_cast<uint32_t>(streams); // ov::optimal_number_of_infer_requests has no negative values
+    } else if (name == ov::streams::num) {
+        const auto streams = engConfig.streamExecutorConfig._streams;
+        return static_cast<int32_t>(streams); // ov::streams::num has special negative values (AUTO = -1, NUMA = -2)
+    } else if (name == ov::affinity) {
+        const auto affinity = engConfig.streamExecutorConfig._threadBindingType;
+        switch (affinity) {
+        case InferenceEngine::IStreamsExecutor::ThreadBindingType::NONE:
+            return ov::Affinity::NONE;
+        case InferenceEngine::IStreamsExecutor::ThreadBindingType::CORES:
+            return ov::Affinity::CORE;
+        case InferenceEngine::IStreamsExecutor::ThreadBindingType::NUMA:
+            return ov::Affinity::NUMA;
+        case InferenceEngine::IStreamsExecutor::ThreadBindingType::HYBRID_AWARE:
+            return ov::Affinity::HYBRID_AWARE;
+        }
+        return ov::Affinity::NONE;
+    } else if (name == ov::inference_num_threads) {
+        const auto num_threads = engConfig.streamExecutorConfig._threads;
+        return num_threads;
+    } else if (name == ov::enable_profiling.name()) {
+        const bool perfCount = engConfig.collectPerfCounters;
+        return perfCount ? "YES" : "NO";
+    } else if (name == ov::hint::inference_precision) {
+        const auto enforceBF16 = engConfig.enforceBF16;
+        return enforceBF16 ? ov::element::bf16 : ov::element::f32;
+    } else if (name == ov::hint::performance_mode) {
+        const auto perfHint = engConfig.perfHintsConfig.ovPerfHint;
+        return perfHint;
+    } else if (name == ov::hint::num_requests) {
+        const auto perfHintNumRequests = engConfig.perfHintsConfig.ovPerfHintNumRequests;
+        return perfHintNumRequests;
+    }
+    // because of some implementation details legacy params still need to be supported even for new API
```

Comment:
Could you please extend the comment for which properties this is required and why?

Reply:
This applies for all the parameters.
I am not sure how we can explain it without the details.
But basically the reason is that we first migrated all the code and all the tests to the new API and only after that we updated plugins to support new configuration API.
So there is period of time when tests used new API but used/use legacy API for configuration.
Later when we clean/correct all the tests we can remove this fallback.

I think from user point of view we expect that when users migrate to the new API they won't use (maybe even must not use) legacy configuration API.

Reply:
Comment rephrased